src/entry_point.py

```diff
@@ -201,6 +256,314 @@ def _constructor_menuinst(prefix, pkg_names=None, root_prefix=None, remove=False
+                    else:
+                        sentinel_str = str(prefix / BIN_DIRECTORY / "conda")
+                    if sys.platform == "win32" and shell != "powershell":
+                        # Remove /cygdrive to make the path shell-independent
+                        sentinel_str = win_path_to_unix(sentinel_str).removeprefix(
+                            "/cygdrive"
+                        )
+                    if any(sentinel_str in match for match in matches):
+                        reverse_plan.append(initializer)
+                        break
+    return reverse_plan
+
+
+def _constructor_uninstall_subcommand(
+    uninstall_dir: str,
+    conda_clean: bool = False,
+    remove_condarcs: str | None = None,
+    remove_caches: bool = False,
+):
+    """
+    Remove a conda prefix or a directory containing conda environments.
+
+    This command also provides options to remove various cache and configuration
+    files to fully remove a conda installation.
+    """
+    from conda.base.constants import PREFIX_MAGIC_FILE
+
+    # See: https://github.com/conda/conda/blob/475e6acbdc98122fcbef4733eb8cb8689324c1c8/conda/gateways/disk/create.py#L482-L488 # noqa
+    ENVS_DIR_MAGIC_FILE = ".conda_envs_dir_test"
+
+    uninstall_prefix = Path(uninstall_dir).expanduser().resolve()
+    if (
+        not (uninstall_prefix / PREFIX_MAGIC_FILE).exists()
+        and not (uninstall_prefix / ENVS_DIR_MAGIC_FILE).exists()
+    ):
+        raise OSError(
+            f"{uninstall_prefix} is not a valid conda environment or environments directory."
+        )
+
+    from shutil import rmtree
+
+    from conda.base.context import context, reset_context
+    from conda.cli.main import main as conda_main
+    from conda.core.initialize import print_plan_results, run_plan, run_plan_elevated
+
+    def _remove_file_directory(file: Path):
+        """
+        Try to remove a file or directory.
+
+        If the file is a link, just unlink, do not remove the target.
+        """
+        try:
+            if not file.exists():
+                return
+            if file.is_dir():
+                rmtree(file)
+            elif file.is_symlink() or file.is_file():
+                file.unlink()
+        except PermissionError:
+            pass
+
+    def _remove_file_and_parents(file: Path, stop_at: Path | None = None):
+        """
+        Remove a file and its parents if empty until reaching the stop_at directory.
+        """
+        if stop_at and not _is_subdir(file, stop_at):
+            raise RuntimeError(f"{file} must be a subdirectory of {stop_at}.")
+        if not file.exists():
+            return
+        _remove_file_directory(file)
+        parent = file.parent
+        while (
+            parent != stop_at and parent.is_dir() and not next(parent.iterdir(), None)
+        ):
+            _remove_file_directory(parent)
+            parent = parent.parent
+
+    def _remove_config_file_and_parents(file: Path):
```

Comment:
There's precedent for users not liking this default behaviour, see https://github.com/conda/conda/issues/13113. Not sure if we should delete empty parents recursively 🤔  Maybe just the config-containing one? Happy to have my mind changed here, though, so let me know.

Reply:
That's what I was intending with `rootdir`, but it was still a little too aggressive. I changed the selection of which directories to remove. The only thing that might be controversial is the `~/.config` directory. I think it's okay to delete if empty. I would at least argue that we should delete it on Windows where configs are stored in different places.